src/Microsoft.OData.Edm/EdmUtil.cs

```diff
@@ -492,6 +500,82 @@ internal static string FullyQualifiedName(IEdmVocabularyAnnotatable element)
             return null;
         }
 
+        internal static string FullyQualifiedName(IEdmAnnotationsTarget annotationsTarget)
+        {
+            string qualifiedName = string.Empty;
+            List<IEdmElement> segments = annotationsTarget.TargetSegments.ToList();
+            IEdmEntityContainer container = segments[0] as IEdmEntityContainer;
+            IEdmEntityContainerElement containerElement = segments[1] as IEdmEntityContainerElement;
+
+            if (container == null || containerElement == null)
+            {
+                return null;
+            }
+
+            qualifiedName = container.FullName() + "/" + containerElement.Name;
+
+            int index = 2;
+
+            IEdmSchemaType schemaType = segments[index] as IEdmSchemaType;
+            IEdmProperty property = segments[index] as IEdmProperty;
+
+            if (schemaType != null)
+            {
+                qualifiedName = qualifiedName + "/" + GetSchemaTypeName(schemaType, segments, index);
+            }
+
+            if (property != null)
+            {
+                qualifiedName = qualifiedName + "/" + GetPropertyName(property, segments, index);

```

Comment:
You could avoid the string allocation done before by changing this to:
```csharp
return string.Contat(container.FullName(), "/", containerElement.Name, "/", GetPropertyName(property, segments, index));
```
i.e., get rid of the `qualifiedName` variable assigned earlier.